src/server/game/Entities/Unit/UnitDefines.h

```diff
@@ -269,25 +269,40 @@ enum MovementFlags : uint32
 
 enum MovementFlags2 : uint32
 {
-    MOVEMENTFLAG2_NONE                     = 0x00000000,
-    MOVEMENTFLAG2_NO_STRAFE                = 0x00000001,
-    MOVEMENTFLAG2_NO_JUMPING               = 0x00000002,
-    MOVEMENTFLAG2_UNK3                     = 0x00000004,        // Overrides various clientside checks
-    MOVEMENTFLAG2_FULL_SPEED_TURNING       = 0x00000008,
-    MOVEMENTFLAG2_FULL_SPEED_PITCHING      = 0x00000010,
-    MOVEMENTFLAG2_ALWAYS_ALLOW_PITCHING    = 0x00000020,
-    MOVEMENTFLAG2_UNK7                     = 0x00000040,
-    MOVEMENTFLAG2_UNK8                     = 0x00000080,
-    MOVEMENTFLAG2_UNK9                     = 0x00000100,
-    MOVEMENTFLAG2_UNK10                    = 0x00000200,
-    MOVEMENTFLAG2_INTERPOLATED_MOVEMENT    = 0x00000400,
-    MOVEMENTFLAG2_INTERPOLATED_TURNING     = 0x00000800,
-    MOVEMENTFLAG2_INTERPOLATED_PITCHING    = 0x00001000,
-    MOVEMENTFLAG2_UNK14                    = 0x00002000,
-    MOVEMENTFLAG2_UNK15                    = 0x00004000,
-    MOVEMENTFLAG2_UNK16                    = 0x00008000
+    MOVEMENTFLAG2_NONE                                  = 0x00000000,
+    MOVEMENTFLAG2_NO_STRAFE                             = 0x00000001,
+    MOVEMENTFLAG2_NO_JUMPING                            = 0x00000002,
+    MOVEMENTFLAG2_UNK3                                  = 0x00000004,        // Overrides various clientside checks
+    MOVEMENTFLAG2_FULL_SPEED_TURNING                    = 0x00000008,
+    MOVEMENTFLAG2_FULL_SPEED_PITCHING                   = 0x00000010,
+    MOVEMENTFLAG2_ALWAYS_ALLOW_PITCHING                 = 0x00000020,
+    MOVEMENTFLAG2_UNK7                                  = 0x00000040,
+    MOVEMENTFLAG2_UNK8                                  = 0x00000080,
+    MOVEMENTFLAG2_UNK9                                  = 0x00000100,
+    MOVEMENTFLAG2_UNK10                                 = 0x00000200,
+    MOVEMENTFLAG2_INTERPOLATED_MOVEMENT                 = 0x00000400,
+    MOVEMENTFLAG2_INTERPOLATED_TURNING                  = 0x00000800,
+    MOVEMENTFLAG2_INTERPOLATED_PITCHING                 = 0x00001000,
+    MOVEMENTFLAG2_UNK14                                 = 0x00002000,
+    MOVEMENTFLAG2_CAN_TRANSITION_BETWEEN_SWIM_AND_FLY   = 0x00004000,
+    MOVEMENTFLAG2_UNK16                                 = 0x00008000
 };
 
+enum UnitMoveType
+{
+    MOVE_WALK = 0,
+    MOVE_RUN = 1,
+    MOVE_RUN_BACK = 2,
+    MOVE_SWIM = 3,
+    MOVE_SWIM_BACK = 4,
+    MOVE_TURN_RATE = 5,
+    MOVE_FLIGHT = 6,
+    MOVE_FLIGHT_BACK = 7,
+    MOVE_PITCH_RATE = 8
+};
+
+#define MAX_MOVE_TYPE     9
```

Comment:
let's move away from defines and go with `static constexpr uint8 MAX_MOVE_TYPE = 9;` instead

Reply:
the old code had `#define MAX_MOVE_TYPE     9` , let's try to keep as few changes as possible in this PR (and in every PR) so we can focus on the actual issues that it fixes

Reply:
so won't do here (but everyone is welcome to improve this in a future PR)